python/samples/concepts/agents/openai_responses/responses_agent_reasoning.py

```diff
@@ -0,0 +1,241 @@
+   - Works for all OpenAI organizations
+   - Reasoning happens internally but no intermediate thoughts are exposed
+   - Still benefits from the model's reasoning process in final responses
+
+2. Reasoning with summary (reasoning={"effort": "high", "summary": "detailed"}):
+   - Requires verified OpenAI organization access
+   - Exposes the model's internal thought process via ReasoningContent
+   - Shows step-by-step reasoning in visual "MODEL THOUGHTS" boxes
+
+The reasoning content shows the internal thought process of models that
+support reasoning (like gpt-5, o3, o1-mini). Examples include both streaming
+and non-streaming invocation patterns with and without tool usage.
+"""
+
+
+class MathPlugin:
+    """A sample Math Plugin used for the concept sample."""
+
+    @kernel_function(description="Add two numbers together")
+    def add(
+        self, a: Annotated[float, "The first number"], b: Annotated[float, "The second number"]
+    ) -> Annotated[float, "The sum of the two numbers"]:
+        result = a + b
+        print(f"Calculator: {a} + {b} = {result}")
+        return result
+
+    @kernel_function(description="Multiply two numbers")
+    def multiply(
+        self, a: Annotated[float, "The first number"], b: Annotated[float, "The second number"]
+    ) -> Annotated[float, "The product of the two numbers"]:
+        result = a * b
+        print(f"Calculator: {a} * {b} = {result}")
+        return result
+
+
+async def create_reasoning_agent_with_summary():
+    """Create a reasoning-enabled agent with summary (requires verified org)."""
+    # Try OpenAI first
+    openai_settings = OpenAISettings()
+    model_id = openai_settings.responses_model_id or openai_settings.chat_model_id
+    if openai_settings.api_key and model_id:
+        client = OpenAIResponsesAgent.create_client()
+        agent = OpenAIResponsesAgent(
+            ai_model_id=model_id,
+            client=client,
+            name="ReasoningAgent",
+            instructions="You are a helpful assistant that thinks step-by-step and uses tools when needed.",
+            plugins=[MathPlugin()],
+            reasoning={"effort": "high", "summary": "detailed"},
+        )
+        return agent, f"OpenAI ({model_id})"
+
+    # Fallback to Azure OpenAI
+    azure_settings = AzureOpenAISettings()
+    if azure_settings.endpoint and azure_settings.responses_deployment_name:
+        client = AzureResponsesAgent.create_client()
+        agent = AzureResponsesAgent(
+            ai_model_id=azure_settings.responses_deployment_name,
+            client=client,
+            name="ReasoningAgent",
+            instructions="You are a helpful assistant that thinks step-by-step and uses tools when needed.",
+            plugins=[MathPlugin()],
+            reasoning={"effort": "high", "summary": "detailed"},
+        )
+        return agent, f"Azure OpenAI ({azure_settings.responses_deployment_name})"
+
+    return None, None
+
+
+async def create_reasoning_agent():
```

Comment:
There's a lot going on in this sample. I think it's totally fine to scope the samples to as small as possible. Meaning, we have one sample for `reasoning_with_summaries` and one sample for `reasoning`. Additionally, we try to break up the `invoke` case versus the `streaming` case.